cpp/src/core/sroptcryst.h

```diff
@@ -1127,6 +1132,34 @@ class srTOptCryst : public srTGenOptElem {
 			*(EPtrs.pEzRe) = (float)(Ety.real());
 			*(EPtrs.pEzIm) = (float)(Ety.imag());
 		}
+        else if(m_uc == 3) //Laue Reflection LD
+		{
+			complex<double> Ehx = sgH[0]*EHSPCs + piH[0]*EHSPCp;
+			complex<double> Ehy = sgH[1]*EHSPCs + piH[1]*EHSPCp;
+			complex<double> Ehz = sgH[2]*EHSPCs + piH[2]*EHSPCp; //Longitudinal component not used (?)
+		
+			//Transverse components of the output electric field in the frame of the output beam:
+			*(EPtrs.pExRe) = (float)(asymFact*Ehx.real());
+			*(EPtrs.pExIm) = (float)(asymFact*Ehx.imag());
+			*(EPtrs.pEzRe) = (float)(asymFact*Ehy.real());
+			*(EPtrs.pEzIm) = (float)(asymFact*Ehy.imag());
+		}
+		else if(m_uc == 4) //Laue Transmission (i.e. Forward Bragg Diffraction (FLD))
+		{
+			//DEBUG
+			//complex<double> DHsgC_p_D0trsC = DHsgC + D0trsC;
+			//complex<double> DHsgCae2_p_D0trsCae2 = DHsgC.real()*DHsgC.real() + DHsgC.imag()*DHsgC.imag() + D0trsC.real()*D0trsC.real() + D0trsC.imag()*D0trsC.imag();
+			//END DEBUG
+
+			complex<double> Etx = m_InvPolTrn[0][0]*E0tSPs + m_InvPolTrn[0][1]*E0tSPp;
+			complex<double> Ety = m_InvPolTrn[1][0]*E0tSPs + m_InvPolTrn[1][1]*E0tSPp;
+		
+			//Transverse components of the output electric field in the frame of the output beam:
+			*(EPtrs.pExRe) = (float)(Etx.real());
+			*(EPtrs.pExIm) = (float)(Etx.imag());
+			*(EPtrs.pEzRe) = (float)(Ety.real());
+			*(EPtrs.pEzIm) = (float)(Ety.imag());
+		}
```

Comment:
This seems to be exactly the same code as for "else if(m_uc == 2)" (?)
Merged it there as: "else if((m_uc == 2) || (m_uc == 4))".